vms/platformvm/txs/builder/builder.go

```diff
@@ -492,6 +531,60 @@ func (b *builder) NewAddValidatorTx(
 	return tx, tx.SyntacticVerify(b.ctx)
 }
 
+func (b *builder) NewAddPermissionlessValidatorTx(
+	stakeAmount,
+	startTime,
+	endTime uint64,
+	nodeID ids.NodeID,
+	rewardAddress ids.ShortID,
+	shares uint32,
+	keys []*secp256k1.PrivateKey,
+	changeAddr ids.ShortID,
+) (*txs.Tx, error) {
+	ins, unstakedOuts, stakedOuts, signers, err := b.Spend(b.state, keys, stakeAmount, b.cfg.AddPrimaryNetworkValidatorFee, changeAddr)
+	if err != nil {
+		return nil, fmt.Errorf("couldn't generate tx inputs/outputs: %w", err)
+	}
+	sk, err := bls.NewSecretKey()
+	if err != nil {
+		return nil, fmt.Errorf("couldn't generate bls key: %w", err)
+	}
```

Comment:
These function shouldn't generate the BLS key. If we actually want to add this function to this interface (which I don't think we should... fwiw) then we should probably just take in the POP.